src/component/environments/EnvironmentActionCell/EnvironmentActionCell.tsx

```diff
@@ -0,0 +1,191 @@
+import { ConditionallyRender } from 'component/common/ConditionallyRender/ConditionallyRender';
+import { IconButton, Tooltip } from '@mui/material';
+import { useNavigate } from 'react-router-dom';
+import AccessContext from 'contexts/AccessContext';
+import { useContext, useState } from 'react';
+import { IEnvironment } from 'interfaces/environments';
+import { formatUnknownError } from 'utils/formatUnknownError';
+import EnvironmentToggleConfirm from '../EnvironmentToggleConfirm/EnvironmentToggleConfirm';
+import EnvironmentDeleteConfirm from '../EnvironmentDeleteConfirm/EnvironmentDeleteConfirm';
+import useEnvironmentApi from 'hooks/api/actions/useEnvironmentApi/useEnvironmentApi';
+import useProjectRolePermissions from 'hooks/api/getters/useProjectRolePermissions/useProjectRolePermissions';
+import { useEnvironments } from 'hooks/api/getters/useEnvironments/useEnvironments';
+import useToast from 'hooks/useToast';
+import { useId } from 'hooks/useId';
+import { useSearchHighlightContext } from 'component/common/Table/SearchHighlightContext/SearchHighlightContext';
+
+interface IEnvironmentTableActionsProps {
+    environment: IEnvironment;
+}
+
+export const EnvironmentActionCell = ({
+    environment,
+}: IEnvironmentTableActionsProps) => {
+    const navigate = useNavigate();
+    const { hasAccess } = useContext(AccessContext);
+    const updatePermission = hasAccess(UPDATE_ENVIRONMENT);
+    const search = useSearchHighlightContext();
+
+    const { setToastApiError, setToastData } = useToast();
+    const { refetchEnvironments } = useEnvironments();
+    const { refetch: refetchPermissions } = useProjectRolePermissions();
+    const { deleteEnvironment, toggleEnvironmentOn, toggleEnvironmentOff } =
+        useEnvironmentApi();
+
+    const [deleteModal, setDeleteModal] = useState(false);
+    const [toggleModal, setToggleModal] = useState(false);
+    const [confirmName, setConfirmName] = useState('');
+
+    const handleDeleteEnvironment = async () => {
+        try {
+            await deleteEnvironment(environment.name);
+            refetchPermissions();
+            setToastData({
+                type: 'success',
+                title: 'Project environment deleted',
+                text: 'You have successfully deleted the project environment.',
+            });
+        } catch (error: unknown) {
+            setToastApiError(formatUnknownError(error));
+        } finally {
+            setDeleteModal(false);
+            setConfirmName('');
+            await refetchEnvironments();
+        }
+    };
+
+    const handleConfirmToggleEnvironment = () => {
+        return environment.enabled
+            ? handleToggleEnvironmentOff()
+            : handleToggleEnvironmentOn();
+    };
+
+    const handleToggleEnvironmentOn = async () => {
+        try {
+            await toggleEnvironmentOn(environment.name);
+            setToggleModal(false);
+            setToastData({
+                type: 'success',
```

Comment:
Not for this PR but at some point is it possible to use an enum/constant/something that a compiler will shout at me for? I spell `success` wrong a lot